src/main/java/spoon/support/reflect/declaration/CtTypeImpl.java

```diff
@@ -894,6 +895,14 @@ public String getQualifiedName() {
 		map(new AllTypeMembersFunction(CtMethod.class)).forEach(new CtConsumer<CtMethod<?>>() {
 			@Override
 			public void accept(CtMethod<?> method) {
+				ClassTypingContext ctc = new ClassTypingContext(CtTypeImpl.this);
+
```

Comment:
> I suggest to move creation of ClassTypingContext above (out of the cycle).

Yes, this was my first move but as I got a lot of errors in tests, I tried inside the loop to be sure of the state of `ClassTypingContext` I will change that :)